### PR TITLE
Full review of repository code by Andrii Slobodianiuk

### PR DESCRIPTION
**Pull request by MaraaaBodnar**

### CODE REVIEW

File: netflix-clone-yt-master/src/app/pages/login/login.component.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+@Component({
+  selector: 'app-login',
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss']
+})
+export class LoginComponent implements OnInit {
+  private router = inject(Router);
```
Comment (MaraaaBodnar) [inline netflix-clone-yt-master/src/app/pages/login/login.component.ts:14]: Prefer using dependency injection in the constructor for better testability and to follow Angular's best practices.

File: netflix-clone-yt-master/src/app/pages/login/login.component.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  ngOnInit(): void {
+    google.accounts.id.initialize({
```
Comment (MaraaaBodnar) [inline netflix-clone-yt-master/src/app/pages/login/login.component.ts:16]: Consider moving hardcoded client_id to environment variables for better security and configurability.

File: netflix-clone-yt-master/src/app/pages/login/login.component.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    google.accounts.id.renderButton(document.getElementById("google-btn"), {
+      theme: 'filled_blue',
+      size: 'large',
+      shape: 'rectangle',
+      width: 350
+    })
+  }
+
+  private decodeToken(token: string){
```
Comment (MaraaaBodnar) [inline netflix-clone-yt-master/src/app/pages/login/login.component.ts:29]: Consider adding a return type to this method for better type safety.

File: netflix-clone-yt-master/src/app/pages/login/login.component.ts (hunk 1 above)
Comment (omruska) [reply netflix-clone-yt-master/src/app/pages/login/login.component.ts:14]: ok👍

File: netflix-clone-yt-master/src/app/pages/login/login.component.ts (hunk 2 above)
Comment (omruska) [reply netflix-clone-yt-master/src/app/pages/login/login.component.ts:16]: 👍do this later

File: netflix-clone-yt-master/src/app/pages/login/login.component.ts (hunk 3 above)
Comment (omruska) [reply netflix-clone-yt-master/src/app/pages/login/login.component.ts:29]: ok👍

**Pull request merged**